packages/catalog-search/package.json

```diff
@@ -45,27 +45,27 @@
   "devDependencies": {
     "@edx/browserslist-config": "1.1.0",
     "@edx/frontend-build": "12.7.0",
-    "@edx/frontend-platform": "4.0.1",
+    "@edx/frontend-platform": "4.6.0",
     "@edx/paragon": "19.25.1",
     "@fortawesome/free-solid-svg-icons": "5.8.1",
-    "@fortawesome/react-fontawesome": "0.1.4",
+    "@fortawesome/react-fontawesome": "0.2.0",
     "@testing-library/jest-dom": "5.11.6",
-    "@testing-library/react": "11.2.6",
+    "@testing-library/react": "12.1.4",
     "@testing-library/react-hooks": "3.4.2",
     "@testing-library/user-event": "13.1.8",
-    "react": "16.14.0",
-    "react-dom": "16.14.0",
+    "react": "17.0.2",
+    "react-dom": "17.0.2",
     "react-instantsearch-dom": "6.8.3",
     "react-router-dom": "5.2.0",
-    "react-test-renderer": "16.14.0"
+    "react-test-renderer": "17.0.2"
   },
   "peerDependencies": {
     "@edx/frontend-platform": "^4.0.1",
     "@edx/paragon": "^19.15.0 || ^20.0.0",
     "@fortawesome/free-solid-svg-icons": "^5.8.1",
-    "@fortawesome/react-fontawesome": "^0.1.4",
-    "react": "^16.12.0",
-    "react-dom": "^16.12.0",
+    "@fortawesome/react-fontawesome": "^0.2.0",
```

Comment:
[curious] Why does this peer dependency need to be updated when `^0.1.4` should already allow `0.2.0`. Is there something in `v0.2.0` that is necessary? Otherwise, this is technically a breaking change for any consumers that install `< 0.2.0`.

Reply:
Mainly `react-fontawesome` now supports forwardRef for version 0.2.x or above which is a breaking change for React version older than 16.3.0. Since we intend to merge this as a breaking change anyways, I updated this as well. Also `"@fortawesome/react-fontawesome": "^0.1.4"` resulted in `Unable to resolve path to module '@fortawesome/react-fontawesome'` which was also resolved with the update.

Reply:
> [clarification] On master, the minimum version of React specified in the repo's peerDependencies is ^16.12.0. Anything to do with React versions older than 16.3.0 should be irrelevant for this repository? For example, if the peer dependency is left as is at ^0.1.4, a consuming MFE like frontend-app-learner-portal-enterprise can still upgrade to ^0.2.0 on their own without causing any issues since its minimum React version is beyond the impacted version.

Yes, versions older than 16.3.0 would be irrelevant for this repo, I may have misinterpreted your first comment, my bad. What you said here makes sense, I have updated the peerDependency for `@fortawesome/react-fontawesome`

> Hm, I can't replicate this issue on master or this PR when installing 0.2.0 in devDeps but left the peer dependency unchanged 🤔

There was a lint failure which you can find in [actions here](https://github.com/openedx/frontend-enterprise/actions/runs/5552827343/jobs/10140713936), thats if the devDep is set to `^0.1.4`, wasn't related to peerDependencies